torchvision/ops/feature_pyramid_network.py

```diff
@@ -80,25 +104,27 @@ def forward(self, x):
         names = list(x.keys())
         x = list(x.values())
 
-        last_inner = self.inner_blocks[-1](x[-1])
+        last_inner = self.last_inner_block(x[-1])
         results = []
-        results.append(self.layer_blocks[-1](last_inner))
-        for feature, inner_block, layer_block in zip(
-            x[:-1][::-1], self.inner_blocks[:-1][::-1], self.layer_blocks[:-1][::-1]
-        ):
-            if not inner_block:
-                continue
-            inner_lateral = inner_block(feature)
-            feat_shape = inner_lateral.shape[-2:]
-            inner_top_down = F.interpolate(last_inner, size=feat_shape, mode="nearest")
-            last_inner = inner_lateral + inner_top_down
-            results.insert(0, layer_block(last_inner))
+        results.append(self.last_layer_block(last_inner))
+
+        i = 0
+        features = x[:-1][::-1]
+        for layer in self.layers:
+            last_inner, result = layer(features[i], last_inner)
+            results.insert(0, result)
+            i += 1
 
         if self.extra_blocks is not None:
             results, names = self.extra_blocks(results, x, names)
 
-        # make it back an OrderedDict
-        out = OrderedDict([(k, v) for k, v in zip(names, results)])
```

Comment:
`OrderedDict` has landed. Would be nice if List Comprehensions worked with iterables...